### PR TITLE
Windows groovy

### PR DESCRIPTION
**Pull request by rasapala**
### 🛠 Summary
JIRA CVS-154689
Add windows builds to CI and on commit.

### CODE REVIEW

File: clean_windows.bat [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
...oolkit\CUDA\v11.8\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\libnvvp;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;S:\Python27;S:\Python27\Scripts;s:\telnet;S:\Tcl\bin;C:\Program Files\SafeNet\LunaClient\win32;C:\Program Files\SafeNet\LunaClient\;C:\ProgramData\chocolatey\bin;C:\Program Files\clinfo\;C:\Program Files\NVIDIA Corporation\Nsight Compute 2022.3.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA cuTENSOR\lib\11.0;C:\Program Files\NVIDIA zlib123dllx64\dll_x64;C:\Program Files\NVIDIA cuDNN\bin;C:\Utils\;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Ninja;C:\Program Files\CMake\bin;C:\Program Files\7-zip;C:\moviTools\21.07.5\win32\bin;C:\Program Files\postgresql\bin;C:\opt\Python39\Scripts\;C:\opt\Python39\;C:\opencl\install\;C:\opencl\;C:\Users\rasapala\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;c:\opt"
```
Comment (mzegla) [inline clean_windows.bat]: That's a huge line that will be hard to analyze if something is wrong. Can we break it to be more readable?
Also lots of hardcoded versions and even `rasapala` occurred in the path 😄

File: build_windows.bat [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
...ance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\devinit;C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x86;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\\MSBuild\Current\Bin;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Utils\;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Ninja;C:\Program Files\CMake\bin;C:\Program Files\7-zip;C:\opt\Python39\Scripts\;C:\opt\Python39\;C:\opencl\install\;C:\opencl\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja
```
Comment (mzegla) [inline build_windows.bat:19]: That's a huge line that will be hard to analyze if something is wrong. Can we break it to be more readable?
Also lots of hardcoded versions. I wonder if we can manage this in a clearer manner.

Also don't you miss closing `"` here?

File: ci/build_test_OnCommit.groovy [hunk 3]
```diff
@@ -33,31 +33,40 @@ pipeline {
...
+            stage("Prepare linux build image") {
+              when { expression { image_build_needed == "true" } }
+                steps {
+                      sh "echo build --remote_cache=${env.OVMS_BAZEL_REMOTE_CACHE_URL} > .user.bazelrc"
+                      sh "make ovms_builder_image RUN_TESTS=0 OV_USE_BINARY=1 BASE_OS=redhat OVMS_CPP_IMAGE_TAG=${shortCommit}"
+                    }
+            }
+            stage('Build windows') {
+              steps {
+                  build job: 'ovms/ovms-windows/'+ env.JOB_BASE_NAME
```
Comment (mzegla) [inline ci/build_test_OnCommit.groovy:66]: Can you make sure indentation is right here?

File: ci/build_test_OnCommitWin.groovy [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    stages {
+        stage ("Clean") {
+            steps {
+                script{
+                    def output1 = bat(returnStdout: true, script: 'clean_windows.bat')
+                }
+            }
+        }
+        // Build windows
+        stage("Build windows") {
```
Comment (mzegla) [inline ci/build_test_OnCommitWin.groovy:17]: Why do we have similar stage in non-windows groovy?

File: clean_windows.bat [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
...ance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\devinit;C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x86;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\\MSBuild\Current\Bin;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Utils\;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Ninja;C:\Program Files\CMake\bin;C:\Program Files\7-zip;C:\opt\Python39\Scripts\;C:\opt\Python39\;C:\opencl\install\;C:\opencl\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja
```
Comment (mzegla) [inline clean_windows.bat:18]: Missing closing `"`?

File: clean_windows.bat (hunk 1 above)
Comment (rasapala) [reply clean_windows.bat]: Removed ununsed.
I will add comment"

File: ci/build_test_OnCommitWin.groovy (hunk 4 above)
Comment (rasapala) [reply ci/build_test_OnCommitWin.groovy:17]: We trigger windows groovy job on jenkins from our oncommit groovy.

**Review by mzegla — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

**Pull request merged**